indico/modules/events/papers/controllers/management.py

```diff
@@ -253,3 +254,45 @@ def _process(self):
             return jsonify_data()
         self.commit = False
         return jsonify_form(form)
+
+
+class RHSetDeadlineBase(RHManagePapersBase):
+    def _process(self):
+        deadline = paper_reviewing_settings.get(self.event_new, self.role)
+        enforce = paper_reviewing_settings.get(self.event_new, 'enforce_{}'.format(self.role))
+        form = DeadlineForm(obj=FormDefaults(deadline=deadline, enforce=enforce), event=self.event_new)
+        if form.validate_on_submit():
+            if self.role == 'layout_reviewer_deadline':
+                text = 'layout reviewing'
+            elif self.role == 'content_reviewer_deadline':
+                text = 'content reviewing'
+            elif self.role == 'judge_deadline':
+                text = 'judging'
+            modified = (paper_reviewing_settings.get(self.event_new, self.role) is not None or
+                        paper_reviewing_settings.get(self.event_new,
+                                                     'enforce_{}'.format(self.role)) is not form.data.get('enforce'))
+            set_deadline(self.event_new, form.data.get('deadline'), self.role, text, form.data.get('enforce'))
+            if modified:
+                flash(_("Paper {} deadline has been modified.".format(text)), 'success')
+            else:
+                flash(_("Paper {} deadline has been set.".format(text)), 'success')
+            return jsonify_data(html=_render_paper_dashboard(self.event_new))
+        return jsonify_form(form)
+
+
+class RHSetContentReviewingDeadline(RHSetDeadlineBase):
+    def _process(self):
+        self.role = 'content_reviewer_deadline'
```

Comment:
This is something you'd set on the class level (then you wouldn't need to override `_process` at all)